### ISSUE TITLE
Adversarial Defense by Restricting the Hidden Space of Deep Neural Networks

### ISSUE THREAD

**Issue by aamir-mustafa** [robust-ml/robust-ml.github.io#6]
**Name**: Adversarial Defense by Restricting the Hidden Space of Deep Neural Networks

**Authors**: Aamir Mustafa, Salman Khan, Munawar Hayat, Roland Goecke, Jianbing Shen, Ling Shao

**Paper**: https://arxiv.org/abs/1904.00887

**Code**: https://github.com/aamir-mustafa/pcl-adversarial-defense

**Venue**: Under Submission

**Does the code implement the robust-ml API and include pre-trained models**: yes

**Dataset**: CIFAR-10

**Threat model**: robust_model.pth.tar , L_inf (Epsilon = 8/255)

**Natural accuracy**: 90.62%

**Claims**:  L_inf (Epsilon = 8/255), 32.32% against PGD

**Comment by anishathalye** [comment]
Thank you for the submission. From a quick glance at your code, I could not find an implementation of the robustml interface. Could you please implement the model, and then we can add your paper to the website?

Also, we require claims to be phrased in terms of the perturbation bound, not based on the particular attack being used (many papers talk about why this is a reasonable thing to do, see e.g. [this](https://github.com/evaluating-adversarial-robustness/adv-eval-paper/) for one explanation).

**Comment by aamir-mustafa** [comment]
Thanks,
I have uploaded the python script, "robust_ml.py" to my Git Repository.

**Comment by anishathalye** [comment]
Merged and deployed in 1757daf213c761291e1c4d443caa3cee3a38f75e.
